packages/react-router/modules/matchPath.js

```diff
@@ -26,15 +26,19 @@ const compilePath = (pattern, options) => {
 /**
  * Public API for matching a URL pathname to a path pattern.
  */
-const matchPath = (pathname, options = {}) => {
+const matchPath = (pathname, options = {}, parentMatch = null) => {
   if (typeof options === 'string')
     options = { path: options }
 
   const { exact = false, strict = false } = options
-  const path = options.path || options.from
+  let path = options.path != null ? options.path : options.from
+  const absolute = isAbsolute(path)
 
-  if (!path)
-    return { url: pathname, isExact: true, params: {} }
+  if (path == null)
+    return { url: parentMatch ? parentMatch.url : '/', isExact: true, params: {} }
+
+  if (!absolute)
+    path = resolvePath(path, parentMatch && parentMatch.url)
```

Comment:
So how should this behave? Suppose I have current URL "/foo" and somewhere the code

```js
<Route path="/nomatch">{() => <Route path="foo"/>}</Route>
```

Am I correct that  reading that the second Route would actually be matching `/foo`, since undefined `base` becomes `/`?

It seems to me that if you have a relative path and the parent did not match, relative URLs should never match?